src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -180,16 +181,15 @@ public FailoverConnectionPlugin(
 
     initSettings();
 
-    this.connectionProvider = connectionProvider;
-
     if (!this.enableFailoverSetting) {
       return;
     }
 
-    AuroraTopologyService topologyService = topologyServiceSupplier.get();
-    topologyService.setPerformanceMetricsEnabled(this.gatherPerfMetricsSetting);
+    this.topologyService = topologyServiceSupplier.get();
```

Comment:
The supplier signature needs to be changed to `Supplier<ITopologyService> topologyServiceSupplier`